src/ConstructiveSolidGeometry/LinePrimitives/Line.jl

```diff
@@ -0,0 +1,74 @@
+struct Line{T,TP,TL} <: AbstractLinePrimitive{T}
+    p1::TP
+    p2::TP
+    linetype::TL
+    function Line( ::Type{T},
+                   p1::Union{CartesianPoint{T}, PlanarPoint{T}},
+                   p2::Union{CartesianPoint{T}, PlanarPoint{T}},
+                   linetype::Union{Val{:inf}, Val{:ray}, Val{:seg}}) where {T}
+        new{T,typeof(p1),typeof(linetype)}(p1, p2)
+    end
+end
+
+function Line( ::Type{T},
+               p1::Union{CartesianPoint{T}, PlanarPoint{T}},
+               p2::Union{CartesianPoint{T}, PlanarPoint{T}}) where {T} #if (inf)Line there are no endpoints
+    Line(T, p1, p2, Val(:inf))
+end
+
+function Ray( ::Type{T},
+               p1::Union{CartesianPoint{T}, PlanarPoint{T}},
+               p2::Union{CartesianPoint{T}, PlanarPoint{T}}) where {T} #if Ray endpoint is p1
+    Line(T, p1, p2, Val(:ray))
+end
+
+function LineSegment( ::Type{T},
+               p1::Union{CartesianPoint{T}, PlanarPoint{T}},
+               p2::Union{CartesianPoint{T}, PlanarPoint{T}}) where {T} #if LineSegment endpoints are p1, and p2
+    Line(T, p1, p2, Val(:seg))
+end
+
+Line(l::Line{T}) where {T} = Line(T, l.p1, l.p2)
```

Comment:
What is this for? 
Isn't that just identity (`Line(l::Line{T}) where {T} = l`)?

Reply:
It is not used right now, but it is not just the identity. It takes any line `Line{T}` which could be of parametric type `Val{:inf}, Val{:ray}, Val{:seg}` and turns it into a line type `Line{T, TP, Val{:inf}}`. This is used eventually for intersections where its convenient to intersect infinite lines instead of segments.

Reply:
Can you remove it here and include it when we need it
(might be that we might not need it at all)